src/odemis/util/transform.py

```diff
@@ -275,6 +302,17 @@ def to_pixel_index(
     array([ 0. ,  0.])
 
     """
+    if isinstance(xy, GeometricTransform):
```

Comment:
If I understood correctly a `GeometricTransform` can be in physical and in pixel space, it just depends on what input it has. Does this code just assume the caller only calls `to_pixel_index` for geometric transforms that had coordinates in physical space as input and vice versa?

Reply:
Typically a `GeometricTransform` is created using its `.from_poinset(src, dst)` method. Consider the following example:
```
src_ji = ...
dst_ji = ...
src_xy = to_physical_space(src_ji, shape, pixel_size)
dst_xy = to_physical_space(dst_ji, shape, pixel_size)
tform_ji = AffineTransform.from_pointset(src_ji, dst_ji)
tform_xy = AffineTransform.from_pointset(src_xy, dst_xy)
```
Then in general `tform_ji` is not equivalent to `tform_xy`. The modified code allows to convert between these two, and indeed assumes you only call `to_physical_space` on something in pixel indices.